code/go/internal/validator/folder_spec.go

```diff
@@ -84,6 +84,13 @@ func (s *folderSpec) validate(packageName string, folderPath string) ve.Validati
 
 		if itemSpec == nil && s.AdditionalContents {
 			// No spec found for current folder item but we do allow additional contents in folder.
+			// Package registry LoadAssets API expects the name not have "-" in it. Check that here
+			// so the issue is caught at validation / lint time of the package.
```

Comment:
No need to mention this here, the spec is the source of truth, it doesn't need to be justified with what the registry does :slightly_smiling_face:

You can though mention that this is done to avoid hyphens in directory names, as by https://github.com/elastic/package-registry/issues/517.

Reply:
Fixed :)